scim/resource_user.go

```diff
@@ -101,12 +109,8 @@ func ResourceUser() common.Resource {
 			if err != nil {
 				return err
 			}
+			setCommonUserFields(d, user, user.UserName)
```

Comment:
We can remove passing user.UserName since we are passing user so inside setCommonUserFields, we can use user.UserName directly.

Reply:
it's required because we need to pass different attributes (UserName for user and ApplicationId for service principal)